sno/working_copy.py

```diff
@@ -897,7 +921,10 @@ def reset(
                                 )
 
                     # Update gpkg_contents
-                    commit_time = datetime.utcfromtimestamp(commit.commit_time)
+                    if commit:
+                        change_time = datetime.utcfromtimestamp(commit.commit_time)
+                    else:
+                        change_time = datetime.utcnow()
```

Comment:
this was set to match the base-commit time, couldn't we keep it like that? Not sure `now()` helps anything?

Reply:
I think this is the only time we ever set `gpkg_contents.last_change`.

> The last_change SHOULD contain the timestamp of when the content in the referenced table was last updated, in ISO8601 format. Note that since it is not practical to ensure that this value is maintained properly in all cases, this value should be treated as informative.


Obviously we can't control what other clients (gdal/qgis/whatever) do, but if we're applying a patch to change the contents of the GPKG then we should be updating the timestamp, I think. We don't have a commit, and the _last_ commit was before the changes were applied.. so i think the only choice left is `datetime.utcnow()`

Reply:
Sure, but we set it to the commit time when you reset/switch/etc. If data changes _after_ that, then the commit time is still relevant. I think we should stick to that principle — so the timestamp of the base commit before applying the un-committed changes from the patch.

Reply:
I don't agree. If it's intended to be used for answering "has this GPKG changed since I last checked it?" then the timestamp needs to be set _after_ any changes were made. If we're applying a patch to the database, but not updating the timestamp, then the timestamp is no longer useful for answering that question. Is there some other question this field is supposed to be used for answering?